### PR TITLE
add more testing scenarios

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**
## Reasoning behind the pull request
- 
- 
- 
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: integrationTests/chainSimulator/vm/esdtImprovements_test.go [hunk 1]
```diff
@@ -1932,3 +1932,238 @@ func TestChainSimulator_SFT_ChangeMetaData(t *testing.T) {
...
+
+	txResult, err := cs.SendTxAndGenerateBlockTilTxIsExecuted(tx, maxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, txResult)
+	require.Equal(t, "success", txResult.Status.String())
+	sftTokenID := txResult.Logs.Events[0].Topics[0]
+
+	log.Info("Issued SFT token id", "tokenID", string(sftTokenID))
+
+	nftMetaData := txsFee.GetDefaultMetaData()
```
Comment (ssd04) [inline integrationTests/chainSimulator/vm/esdtImprovements_test.go:1995]: rename to sft meta data? or just to metaData

Comment (ssd04) [reply integrationTests/chainSimulator/vm/esdtImprovements_test.go:1995]: same for the other tests

Comment (BeniaminDrasovean) [reply integrationTests/chainSimulator/vm/esdtImprovements_test.go]: Renamed

**Review by sstanculeanu — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by sasurobert — DISMISSED** [review]

File: integrationTests/chainSimulator/vm/esdtImprovements_test.go [hunk 2]
```diff
@@ -1932,3 +1932,238 @@ func TestChainSimulator_SFT_ChangeMetaData(t *testing.T) {
...
+		HasValue: true,
+		Value:    20,
+	}
+
+	activationEpochForSaveToSystemAccount := uint32(2)
+	activationEpochForDynamicNFT := uint32(4)
+
+	numOfShards := uint32(3)
+	cs, err := chainSimulator.NewChainSimulator(chainSimulator.ArgsChainSimulator{
+		BypassTxSignatureCheck:   false,
```
Comment (ssd04) [inline integrationTests/chainSimulator/vm/esdtImprovements_test.go:2091]: we need to set `BypassTxSignatureCheck` to `true` after last merged changes from here https://github.com/multiversx/mx-chain-go/pull/6268

**Review by ssd04 — APPROVED** [review]

File: integrationTests/chainSimulator/vm/esdtImprovements_test.go (hunk 2 above)
Comment (BeniaminDrasovean) [reply integrationTests/chainSimulator/vm/esdtImprovements_test.go]: Done

**Review by sstanculeanu — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**